plugins/modules/zos_volume_init.py

```diff
@@ -0,0 +1,249 @@
+    required: true
+    type: str
+  verify_volid:
+    description:
+      - Verify that the volume serial matches what is on the existing volume or minidisk.
+      - I(verify_volid) must be 1 to 6 alphanumeric characters or "*NONE*".
+      - To verify that a volume serial number does not exist, use I(verify_volid="*NONE*").
+      - If I(verify_volid) is specified and the volume serial number does not match that found on the volume or minidisk, initialization does not complete.
+      - If I(verify_volid="*NONE*") is specified and a volume serial is found on the volume or minidisk, initialization does not complete.
+      - Note - This option is NOT a boolean, please leave it blank in order to skip the verification. # TODO
+    required: false
+    type: str
+  verify_offline:
+    description:
+      - Verify that the device is not online to any other systems, initialization does not complete.
+      - Beware, defaults set on target z/OS systems may override ICKDSF parameters. # TODO
+    type: bool
+    required: false
+    default: true
+  volid:
+    description:
+      - The volume serial number used to initialize a volume or minidisk.
+      - Expects 1-6 alphanumeric, national ($, \\#, \\@) or special characters.
+      - A I(volid) with less than 6 characters will be padded with spaces.
+      - Characters are not validated so check with operating system guide for valid alphanumeric characters.
+      - A I(volid) can also be referred to as volser or volume serial number.
+      - When I(volid) is not specified for a previously initialized volume or minidisk, the volume serial number will remain unchanged.
+    required: false
+    type: str
+  vtoc_size:
+    description:
+      - The number of tracks to initialize the volume table of contents (VTOC) with.
+      - The VTOC will be placed in cylinder 0 head 1.
+      - If no tracks are specified it will default to the number of tracks in a cylinder minus 1. Tracks in a cylinder vary based on direct-access storage
+        device (DASD) models, for 3390 a cylinder is 15 tracks.
+    required: false
+    type: int
+  index:
+    description:
+      - Create a volume table of contents (VTOC) index.
+      - The VTOC index enhances the performance of VTOC access.
+      - When set to C(false), no index will be created.
+    required: false
+    type: bool
+    default: true
+  sms_managed:
+    description:
+      - Specifies that hte volume be managed by Storage Management System (SMS).
+      - If I(sms_managed) is C(true) then I(index) must also be C(true).
+    type: bool
+    required: false
+    default: true
+  verify_volume_empty:
+    description:
+      - Verify that no data sets other than the volume table of contents (VTOC) index or the VSAM Volume Data Set(VVDS) exist on the target volume.
+      - Beware that z/OS system defaults can override ICKDSF parameters.
+    required: false
+    type: bool
+    default: true
+"""
```

Comment:
Might want to now add a `seealso` section if we mention the zos_backup_restore module.
```
seealso:
- module: zos_backup_restore
```